### PR TITLE
expression: fix the behavior of `DATE_SUB`

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

fix `date_sub` when its output should be `Time`.
And try to open window function as default.

### What is changed and how it works?

Add three new signatures.

### Check List 

Tests 

 - Unit test (will add it later)

### CODE REVIEW

File: expression/builtin_time.go [hunk 1]
```diff
@@ -2760,6 +2760,18 @@ func (du *baseDateArithmitical) addDuration(ctx sessionctx.Context, d types.Dura
 	return retDur, false, nil
 }
 
+func (du *baseDateArithmitical) subDuration(ctx sessionctx.Context, d types.Duration, interval string, unit string) (types.Duration, bool, error) {
```
Comment (lamxTyler) [inline expression/builtin_time.go]: Can you add some unit tests? It will affect the test coverage.

**Comment by zz-jason** [comment]
We need to cherry-pick this commit to release-2.1 as well.

**Pull request merged**